Added some missing options

GastonZalba opened the pull request:
Shared options
- videoStabilization
- colorEffect
- dynamicRange
- imageEffectMode
- statistics
- meteringMode
- flickerMode
- roi
- awbGains
- annotate
- annotateExtra

Still options
- raw
- quality
- thumbnail
- burst
- exif
- gpsExif

Added HQ Camera mode

GastonZalba committed:
Added missing options and improved tests

GastonZalba committed:
Updated README

GastonZalba committed:
Added more missing options and HQ Camera mode

GastonZalba committed:
Fix roi argument

gregnr commented on src/lib/shared-args.ts:
```diff
@@ -85,14 +85,85 @@ export function getSharedArgs(options: StillOptions | StreamOptions): string[] {
+     * the U and Y channels of the image. For example, --colfx 128:128
+     * should result in a monochrome image.
+     */
+    ...(options.colourEffect ? ['--colfx', options.colourEffect.join(':')] : []),
```
Can we change this to `colorEffect`? I personally spell it "colour" as well, but I'd prefer to keep as standard as possible which (for better or for worse) is more frequently the American version.

GastonZalba replied:
yes

gregnr commented on src/lib/shared-args.ts:
```diff
@@ -85,14 +85,85 @@ export function getSharedArgs(options: StillOptions | StreamOptions): string[] {
+     * Video Stabilisation
+     * In video mode only, turn on video stabilization.
+     */
+    ...(options.videoStabilisation ? ['--vstab'] : []),
```
Same here: `videoStabilization`

gregnr commented on src/index.ts:
```diff
@@ -44,3 +44,47 @@ export enum AwbMode {
   GreyWorld = 'greyworld',
 }
+
+export enum ImxfxMode {
```
Worth naming `ImageEffectMode` to match arg name?

GastonZalba replied:
yes

gregnr commented on src/index.ts:
```diff
@@ -44,3 +44,47 @@ export enum AwbMode {
+export enum FlickerMode {
+  Off = 'off',
+  Auto = 'auto',
+  '50hz' = '50hz',
+  '60Hz' = '60hz',
```
Worth prefixing with some letters like `Frq50Hz` so that devs don't have to access via `FlickerMode['50hz']` syntax?

GastonZalba replied:
yes

gregnr commented on README.md:
```diff
@@ -307,8 +322,17 @@ const streamCamera = new StreamCamera({
+- `videoStabilisation: boolean` - _Default: `false`_
+- [`meteringMode`](#meteringMode) - _Default: `MeteringMode.Off`_
+- [`flickerMode`](#flickerMode) - _Default: `null`_
+- `roi: [number, number, number, number]` - _Default: `null`_
```
Worth mentioning default when `undefined` means full sensor region?

GastonZalba replied:
Changed null to _Default: Full sensor_

gregnr commented on README.md:
```diff
@@ -307,8 +322,17 @@ const streamCamera = new StreamCamera({
+- [`dynamicRange: DynamicRange`](#dynamicrange) - _Default: `DynamicRange.Off`_
+- `videoStabilisation: boolean` - _Default: `false`_
+- [`meteringMode`](#meteringMode) - _Default: `MeteringMode.Off`_
+- [`flickerMode`](#flickerMode) - _Default: `null`_
```
Is this off by default? Could we mention that here?

GastonZalba replied:
It's off by default. Yes

gregnr reviewed (changes requested):
This is really great @GastonZalba! Thanks for your work on this, and sorry for the delay. I just a few DX things I would love to get your thoughts on before merging.

GastonZalba commented:
Hi, excelent. I'm going to look at this and come back. 

Another thing that would seem appropriate to do (i can make a separate PR) would be to move all the enums to a separate file and _then_ reexport them from `index.ts`. So, the current behaviour would be the same, but it will allow to import only the enums (using something like `pi-camera-connect/dist/@enums`) from a frontend appplication (like react/electron) without requiring the entire library and preventing the fatal import errors from `child_process` and `stream` dependencies.

Sorry if my english is a bit broken.

GastonZalba committed:
Requested changes applied

GastonZalba commented:
@gregnr changes applied. Let me know what you think about the previous enums suggestion (move them to a separate file). 
I'm also working on enabling the preview window and arguments (`fullscreen`, `preview`, `keypress` and `dispnum`). I'll be back in a few days with that.

gregnr commented on src/lib/shared-args.ts:
```diff
@@ -85,14 +85,85 @@ export function getSharedArgs(options: StillOptions | StreamOptions): string[] {
+    ...(options.flickerMode ? ['--flicker', options.flickerMode] : []),
+
+    /**
+     * Video Stabilisation
```
```suggestion
     * Video Stabilization
```

gregnr committed:
chore: stabilization comment spelling

GastonZalba committed:
Added annotate and exif. Improved readme

GastonZalba committed:
Merge branch 'master' of https://github.com/GastonZalba/pi-camera-connect

GastonZalba committed:
README details and format

GastonZalba commented:
@gregnr Hi, i finally added a few more options (`annotate`, `exif` and related), expanded a little the still test, and cleaned up some mess that i left after the last requested changes. In the README I also modified some values that I had set as "default" from `null` to `undefined`, which I understand is more correct (and added some missing info as well). Let me know if any further adjustment is needed

GastonZalba committed:
Fix uppercase in FlickerMode enum
